editoast/src/schema/rolling_stock/mod.rs

```diff
@@ -289,10 +298,18 @@ mod tests {
     }
 
     #[test]
-    fn test_de_effort_curve_unvalid() {
+    fn test_de_effort_curve_invalid_due_to_mismatched_lengths() {
         assert!(
             serde_json::from_str::<EffortCurve>(r#"{ "speeds": [0, 1], "max_efforts": [] }"#)
                 .is_err()
         );
     }
+
+    #[test]
+    fn test_de_effort_curve_invalid_due_to_duplicates_coordinates() {
+        assert!(serde_json::from_str::<EffortCurve>(
+            r#"{ "speeds": [0, 1, 1], "max_efforts": [1, 1, 1] }"#
```

Comment:
```suggestion
            json!({ "speeds": [0, 1, 1], "max_efforts": [1, 1, 1] })
```

Reply:
```suggestion
            json!({ "speeds": [0, 1, 1], "max_efforts": [5, 4, 3] })
```

We can also add tests on with negative speeds & max_effort.